src/infra/flow-system-inmem/tests/tests/test_flow_service_inmem.rs

```diff
@@ -414,6 +426,293 @@ async fn test_manual_trigger() {
+              "foo" HardCompaction:
+                Flow ID = 2 Waiting AutoPolling Schedule(wakeup=160ms)
+                Flow ID = 1 Finished Success
+                Flow ID = 0 Finished Success
+
+            #9: +110ms:
+              "bar" HardCompaction:
+                Flow ID = 3 Finished Success
+              "foo" HardCompaction:
+                Flow ID = 2 Waiting AutoPolling Schedule(wakeup=160ms)
+                Flow ID = 1 Finished Success
+                Flow ID = 0 Finished Success
+
+            #10: +160ms:
+              "bar" HardCompaction:
+                Flow ID = 3 Finished Success
+              "foo" HardCompaction:
+                Flow ID = 2 Waiting AutoPolling Executor(task=3, since=160ms)
+                Flow ID = 1 Finished Success
+                Flow ID = 0 Finished Success
+
+            "#
+        )
+    );
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[test_log::test(tokio::test)]
+async fn test_manual_trigger_compaction_with_config() {
+    let max_slice_size = 1_000_000u64;
+    let max_slice_records = 1000u64;
+    let harness = FlowHarness::new();
+
+    let foo_id = harness.create_root_dataset("foo").await;
+
+    harness.eager_dependencies_graph_init().await;
+    harness
+        .set_dataset_flow_compaction_rule(
+            harness.now_datetime(),
+            foo_id.clone(),
+            DatasetFlowType::HardCompaction,
+            CompactionRule::new_checked(max_slice_size, max_slice_records).unwrap(),
+        )
+        .await;
+
+    let foo_flow_key: FlowKey =
+        FlowKeyDataset::new(foo_id.clone(), DatasetFlowType::HardCompaction).into();
+
+    let test_flow_listener = harness.catalog.get_one::<FlowSystemTestListener>().unwrap();
+    test_flow_listener.define_dataset_display_name(foo_id.clone(), "foo".to_string());
+
+    // Remember start time
+    let start_time = harness
+        .now_datetime()
+        .duration_round(Duration::try_milliseconds(SCHEDULING_ALIGNMENT_MS).unwrap())
+        .unwrap();
+
+    // Run scheduler concurrently with manual triggers script
+    tokio::select! {
+        // Run API service
+        res = harness.flow_service.run(start_time) => res.int_err(),
+
+        // Run simulation script and task drivers
+        _ = async {
+                // Task 0: "foo" start running at 10ms, finish at 20ms
+                let task0_driver = harness.task_driver(TaskDriverArgs {
+                    task_id: TaskID::new(0),
+                    dataset_id: Some(foo_id.clone()),
+                    run_since_start: Duration::try_milliseconds(10).unwrap(),
+                    finish_in_with: Some((Duration::try_milliseconds(10).unwrap(), TaskOutcome::Success(TaskResult::Empty))),
+                    max_slice_size: Some(max_slice_size),
+                    max_slice_records: Some(max_slice_records),
+                });
+                let task0_handle = task0_driver.run();
```

Comment:
Why would task start before the trigger? If trigger is at time 40ms, the task should go after, maybe at 50ms.

Reply:
Fixed